### DISCUSSION TITLE
Need help translating meeting finder interface into Japanese

### DISCUSSION THREAD

**Discussion by Hiro-Tokyo** [code4recovery/12-step-meeting-list#975] [Q&A] [answered]
Hello team! 

We, Japan GSB and Japan GSB technology committee are working with US/Canada GSO to translate Meeting Guide App user interface into Japanese. Also we would like to have Japanese user interface of 12 Step Meeting List plug-in into Japanese.

Is there any better way to translate meeting finder interface into Japanese?

Would you help/support us?

Hiroyuki K. - Japan GSB technology committee

**Comment by joshreisner** [comment, marked as answer]
hi Hiro! we'll be pleased to help. do you have a demo website set up yet? 

if not, we can help set one up

**Reply by joshreisner** [reply]
hi @Hiro-Tokyo just wanted to make sure you saw my message about a demo website. that's an important first step so we can get a shared picture of what needs to be done

**Reply by Hiro-Tokyo** [reply]
Hi Josh, sorry for my late reply. I was talking with Doug and Nori.

We would like to ask you to built demo site. We do not have demo site yet. And our opinion is: asking you and the code for recovery team to help us launching demo site is best way for start.

Could you do it? Please let us know how we should do.

**Reply by joshreisner** [reply]
Hi @Hiro-Tokyo , we can help you solve problems you encounter with the meeting finder specifically. Sounds like your group probably wants to hire a contractor to design and build the rest of the website. But this can be a parallel effort; while you (and they) work on designing a website, we can help get the meeting finder ready.

Couple questions:

1. what will the website URL be?
2. will the whole website need to be multilingual, or will it be primarily Japanese?
3. will the meetings listed be primarily in Japanese, English, or both? 
4. do you have meeting data ready yet in a spreadsheet or database?

**Reply by Hiro-Tokyo** [reply]
Thanks Josh!

1. We will let you know about the URL of demo website. Please let us take time to ready.
2. It's primarily Japanese as central offices/intergroups of Japanese meetings will use it.
3. It's primarily Japanese.
***Is there any way to switch language?
4. Not yet. Do we need to real meetings list or some sample meetings list will work?

Japan Online Meeting Intergroup is using the plug-in but it is running so we think we can not use it for test.

**Reply by joshreisner** [reply]
ok thanks @Hiro-Tokyo !

one way that might make it easy for us to collaborate would be by using a Google Sheet so that we can both make edits to the data

I will
* create a demo google sheet using sample data
* set up an all-japanese meeting finder page using TSML UI (this can be embedded on any website, not just wordpress)
* use Google Translate at first, then you can suggest where it needs to be fixed

this approach will work with Meeting Guide

the interface will be in japanese, english, spanish, or french, depending on what language your browser is set to use

but the data (meeting names and addresses) will all be in japanese

as you develop your list of meetings you can gradually replace what's in the google sheet with real data

let me know if that sounds ok!

**Reply by Hiro-Tokyo** [reply]
Thanks Josh! Your suggestion would be helpful! Please go ahead.
Just in case, I would like to make clear about a point. If Japan GSB tech team need to launch demo website page, please wait for a while, just after Japan GSB meeting on the start of December. But will you set up the demo page?

**Reply by joshreisner** [reply]
ok sounds good. i'm going away for a US holiday for two weeks, but i will pick this up again in early december.

**Reply by Hiro-Tokyo** [reply]
Thanks @joshreisner! Enjoy your holiday!

**Reply by joshreisner** [reply]
hi @Hiro-Tokyo !

i've added japanese language support now to TSML UI.

if your browser is set to use japanese, then you should see the interface in japanese. for example: https://tsml-ui.code4recovery.org/tests/aasanjose.html

![Screenshot 2022-12-03 at 16-46-43 ミーティング](https://user-images.githubusercontent.com/1551689/205468437-df0c6559-6c46-44ca-9eac-6ea060ea4db8.png)

i used google translate and may have gotten some things wrong. feel free to suggest new translations: https://github.com/code4recovery/tsml-ui/blob/main/src/i18n/ja.ts

in addition to the interface, TSML UI now accepts japanese-language _data_. i set up an example site with two meetings: https://tsml-ui.code4recovery.org/tests/japan.html. entries in [this google sheet](https://docs.google.com/spreadsheets/d/1S18u8LyDXJy-nkDbAi321imElV8JNXR3R8ho0P0srHE/edit#gid=0) will populate that meeting finder. if you want to make edits, please request access.

![Screenshot 2022-12-03 at 16-56-06 ミーティング](https://user-images.githubusercontent.com/1551689/205468628-59818929-74d3-4ce9-a0ea-d371c4235d9b.png)

this means that the days of the week, meeting types, and addresses can all be entered in japanese and should just work.

please let me know if there's stuff here i can clarify, and/or how you'd like to proceed.

**Reply by Hiro-Tokyo** [reply]
Thank you Josh for your amazing work! I will have some suggestion to change translation. 

I would like to make sure that I can use 'propose changes' function of the file. Is it right way? 

Also I would like to try to edit the google sheet!

**Reply by joshreisner** [reply]
Hi Hiro, yes, if you want to edit file > propose changes that is a good way. 

Looks like you added a few meetings! Let me know if you have any questions about that.

I tried a new thing with the `Regions` column, where the geocoding handles it automatically. Let me know if that isn't working for you, we can change or remove it.

**Reply by Hiro-Tokyo** [reply]
Thanks! I will do that way. Please let me take time a bit.

**Reply by Hiro-Tokyo** [reply]
I have a question. When these column are used? 

email_edit_url:
email_public_url:
email_subject:

**Reply by Hiro-Tokyo** [reply]
And 'platform' is bus stop or train station?

**Reply by joshreisner** [reply]
hi Hiro, those three "email" strings are used for sending feedback emails. this is an optional button that appears on meetings that says "Update Meeting Info" -- you can see an example here: https://sussexaa.org/meetings/?meeting=early-birds-12-12-inside

right now the button is not enabled, and if you choose not to enable it you won't see the strings

when we use "Address / Platform" we mean "platform" in the sense of Zoom, GoToMeeting, Skype, etc

**Reply by Hiro-Tokyo** [reply]
I sent my suggestions. Plus, I may have some small changes but if you are okay for it, please go ahead with my current request. Our team would like to review it.

**Reply by joshreisner** [reply]
ok great thanks! your suggestions are now live, so if you click refresh at https://tsml-ui.code4recovery.org/tests/japan.html you should see them now 👍

**Reply by Hiro-Tokyo** [reply]
Thank you Josh! Our team will review it. When I would like to change Japanese translation, can I take same way?

**Reply by joshreisner** [reply]
sure!

**Reply by Hiro-Tokyo** [reply]
'Near Me' shows as mi but km is familiar in Japan so is there any way to change to km?
<img width="506" alt="Screen Shot 2022-12-07 at 20 43 59" src="https://user-images.githubusercontent.com/100449629/206171289-18904caf-1284-429c-b52b-676fc68221dc.png">

**Reply by joshreisner** [reply]
Oops yes it is switched now 👍

**Reply by Hiro-Tokyo** [reply]
Thanks!

**Reply by Hiro-Tokyo** [reply]
Hi Josh, I'm sorry for not having our updates long time. I have a question. When we switch user interface to TSML from legacy in 12 step meeting list plug-in of Wordpress, Japanese interface will be showing up on website automatically? Japanese online intergroup is using the plug-in.

**Reply by joshreisner** [reply]
yes i think it should be there if you switch to TSML UI

**Reply by Hiro-Tokyo** [reply]
Thanks! I will try it!

**Reply by Hiro-Tokyo** [reply]
I checked it with the plug-in. When we change the setting to TSML UI from Legacy UI, meeting type is still staying in English. For example, English will change to 英語 (English in Japanese) automatically with direct CSV importing on test site which you made but it does not happen on Japan Online Intergroup Website with the plug-in. Any solution?

**Reply by joshreisner** [reply]
🤔 hm, what's the website URL?

**Reply by joshreisner** [reply]
also, did you want the distance units `mi` and `km` to be translated in japanese? i just noticed they are not

**Reply by Hiro-Tokyo** [reply]
Here's the website URL using Legacy UI at the moment. I checked it (just check) then backed in Legacy UI. Do I need to change again? If so, please let me know. I can change the setting for you when you can.

https://aajapan-online-intergroup.org/index.php/meetings/

Also if you can change unit to Japanese, it would be helpful.

mi -> マイル
km -> キロメートル

**Reply by joshreisner** [reply]
hi hiro,

is there a shorter way to express km in japanese with fewer characters? note that space is limited so we don't spell out "miles" and "kilometers" in english. 

<img width="161" alt="Screenshot 2023-02-26 at 8 50 29 AM" src="https://user-images.githubusercontent.com/1551689/221424411-bcc25eea-cd10-45b5-916a-e7d2b58a2458.png">

could we say 4.2キロ?

re the types, the problem is that we get the types from the plugin, and it's not yet translated. so the best thing for now is to customize them. can you add this to your theme's `functions.php`?

```
if (function_exists('tsml_custom_types')) {
    tsml_custom_types([
        '11' => 'ステップ11 黙想',
        '12x12' => '12のステップと12の伝統',
        'active' => 'アクティブ',
        'AL-AN' => 'アラノンと同時進行',
        'A' => '無宗教',
        'ABSI' => 'ビルはこう思う',
        'AL' => 'アラティーンと同時進行',
        'ASL' => 'アメリカ手話',
        'B' => 'ビッグブック',
        'BA' => 'ベビーシッターあり',
        'BE' => 'ニューカマー',
        'BRK' => '朝食',
        'BI' => 'バイセクシャル',
        'C' => 'クローズド',
        'CAN' => 'キャンドル',
        'CF' => 'お子さま歓迎',
        'D' => 'ディスカッション',
        'DB' => '電子献金',
        'DD' => '重複診断',
        'DR' => '今日を新たに',
        'EN' => '英語',
        'FF' => '香水なし',
        'FR' => 'フランス語',
        'G' => 'ゲイ',
        'GR' => 'グレープバイン',
        'H' => 'バースデー',
        'HE' => 'ヘブライ語',
        'ITA' => 'イタリア語',
        'JA' => '日本語',
        'KOR' => '韓国語',
        'L' => 'レズビアン',
        'LGBTQ' => 'LGBTQ',
        'LIT' => '書籍',
        'LS' => 'リビングソーバー',
        'M' => '男性',
        'MED' => '黙想',
        'N' => 'ネイティブアメリカン',
        'NDG' => '先住民',
        'O' => 'オープン',
        'OUT' => 'アウトドア',
        'P' => '職業人',
        'POC' => '有色人種',
        'POL' => 'ポーランド語',
        'POR' => 'ポルトガル語',
        'PUN' => 'パンジャブ語',
        'RUS' => 'ロシア語',
        'S' => 'スペイン語',
        'SEN' => '高齢者',
        'SM' => '喫煙可',
        'SP' => 'スピーカー',
        'SPD' => 'スピーカー/ディスカッション',
        'ST' => 'ステップスタディ',
        'T' => 'トランスジェンダー',
        'TC' => '一時的休止中',
        'TR' => '伝統',
        'W' => '女性',
        'X' => '車いすアクセス',
        'XB' => '車いす使用者用トイレ',
        'XT' => 'クロストーク可能',
        'Y' => 'ヤング',
    ]);
}
```

**Reply by Hiro-Tokyo** [reply]
Thanks Josh! 

Yes, キロ for km would be fine so please go ahead!

I will check with your idea then get you back!

**Reply by joshreisner** [reply]
thanks hiro. should there be a space between the number and the characters, or no? thx

**Reply by Hiro-Tokyo** [reply]
We don't use space usually so no space would be fine.

And description of open and closed would be in Japanese on question symbol? If I need to put additional code on functions php, please let me know.
<img width="229" alt="Screenshot 2023-02-27 at 14 01 23" src="https://user-images.githubusercontent.com/100449629/221478376-1c734514-aedd-4f47-9924-47278e28f35a.png">

**Reply by Hiro-Tokyo** [reply]
Your idea worked! So please let me know about description part.
<img width="366" alt="Screenshot 2023-02-27 at 14 06 57" src="https://user-images.githubusercontent.com/100449629/221479129-ddee4aae-6e75-478e-be7f-26030dcb3d42.png">

**Reply by joshreisner** [reply]
hm, for some reason i'm having a hard time recreating this (i'm seeing japanese) -- i'll have to research it more…

![Screenshot 2023-02-28 at 8 46 30 AM](https://user-images.githubusercontent.com/1551689/221920624-215785b8-7d69-4527-b88c-f5d81787b73d.png)

**Reply by Hiro-Tokyo** [reply]
Thank you Josh for your hard work. Please let me know if I can support!

**Reply by Hiro-Tokyo** [reply]
In addition, this issue happens on just plug-in , not on Google Sheet (direct).

**Reply by joshreisner** [reply]
hi hiro,

try adding this to your theme's `functions.php`:

```php
if (!empty($tsml_programs)) {
    $tsml_programs['aa']['type_descriptions'] = [
        'C' => 'クローズド・ミーティングはＡＡメンバー、もしくはお酒に問題がある、お酒をやめたいと願っている方が参加できます。',
        'O' => 'オープン・ミーティングはアルコホーリクス・アノニマスに興味のある方ならどなたでもご参加いただけます。アルコール依存症でない方はオブザーバーとしてのみのミーティング参加になる可能性があります。',
    ];
}
```

**Reply by Hiro-Tokyo** [reply]
Worked! Thank you Josh!
<img width="633" alt="Screenshot 2023-03-01 at 23 24 14" src="https://user-images.githubusercontent.com/100449629/222189593-0f8cc05d-7762-4f97-b0f0-67580db03c54.png">
<img width="733" alt="Screenshot 2023-03-01 at 23 23 59" src="https://user-images.githubusercontent.com/100449629/222189611-1db3c63c-c60a-4c06-ae47-9ca39397c60c.png">

**Reply by Hiro-Tokyo** [reply]
Hi Josh, I talked Kansai area central office website team in Japan. They would like to test TSML UL for their website. They are considering to build new website. How can I share google sheet you made? And I will ask you to make google sheet direct embed code for them.

https://docs.google.com/spreadsheets/d/1S18u8LyDXJy-nkDbAi321imElV8JNXR3R8ho0P0srHE/edit#gid=0

https://tsml-ui.code4recovery.org/

**Reply by joshreisner** [reply]
hi hiro! tell them to go to this sheet and duplicate it, and then send me the address. also ask them to go to mapbox.com and signup for an account and create an access token, and give me that too. (both steps are 100% free/no-cost)

**Reply by joshreisner** [reply]
the sheet permissions should be: only they can edit, but anyone with the link can view

**Reply by Hiro-Tokyo** [reply]
Thanks Josh. Do we need to send you google sheet api key also?

**Reply by joshreisner** [reply]
sure that would be great. i mean once you have all that stuff, you can pretty much just assemble it from there. but yes if you want to send it to me i can confirm

```html
<div
  id="tsml-ui"
  data-src="https://docs.google.com/spreadsheets/d/your.google.sheet.id/edit#gid=more.stuff.here"
  data-google="your.google.api.key"
  data-mapbox="pk.your.mapbox.access.token.goes.here"
  data-timezone="Asia/Tokyo"
></div>
<script src="https://tsml-ui.code4recovery.org/app.js" async></script>
```

**Reply by Hiro-Tokyo** [reply]
Hi Josh, it worked and test is going well. One issue is distance is showing as mile so any solution?

Test site is:
https://aa-kansai-hp.jimdofree.com/%E3%83%9F%E3%83%BC%E3%83%86%E3%82%A3%E3%83%B3%E3%82%B0%E3%82%AC%E3%82%A4%E3%83%89%E3%83%86%E3%82%B9%E3%83%88/

**Reply by Hiro-Tokyo** [reply]
<img width="534" alt="Screenshot 2023-03-05 at 21 29 05" src="https://user-images.githubusercontent.com/100449629/222968477-dcd8ed47-1d20-45d8-abae-41e969ec9cd3.png">

**Reply by kiyote33** [reply]
Hi @Hiro-Tokyo , I think Josh may be sleeping in today so I will try to assist. There are a couple settings which may affect whether Distance is calculated in Miles of Kilometers. The first one is our **Meetings/Settings/General/Distance Units** which affects distance settings on our Meeting List page. The second distance setting comes on the the Mapping page when you "Get Directions". There the Options default setting for Distance Units is "Automatic", but you can override that selection by choosing "miles" or "km". 

Hope this helps...

**Reply by joshreisner** [reply]
Thanks Wayne for covering for me! Hiro's actually using TSML UI without TSML so he'll a javascript fix. Hiro, add this before the code I gave you:

```html
    <script>
      var tsml_react_config = {
        distance_unit: 'km',
      };
    </script>
```

while i have you both here, can you open your browser javascript console and type `navigator.languages` and tell me what it says? i see `[ "en-US", "en"]`. the reason i ask is i am wondering if we can make this setting automatic - only show miles to users who have `en-US` in the array. since you both are english speaking but not in the USA i'll be interested to see what you have

**Reply by kiyote33** [reply]
Hi Josh, I get **['en-US', 'en', 'fr', 'mt']** in the javascript console.

**Reply by Hiro-Tokyo** [reply]
I get  ['ja', 'en-US', 'en']
<img width="621" alt="Screenshot 2023-03-06 at 9 56 37" src="https://user-images.githubusercontent.com/100449629/222996989-95236d98-0fee-45d4-ac84-f2463a54f030.png">

**Reply by Hiro-Tokyo** [reply]
And I'm asking them to add the code you gave me.

**Reply by joshreisner** [reply]
oh ok thanks! i don't think my plan will work then 😢 

hiro, to fix that paypal warning, it should just be `tokyoyoungpeople` without the rest of the URL. we changed it recently so it will work properly in the meeting guide app

**Reply by Hiro-Tokyo** [reply]
paypal warning was improved!
<img width="578" alt="Screenshot 2023-03-06 at 11 17 39" src="https://user-images.githubusercontent.com/100449629/223004608-26a97ab5-9955-4758-93b6-d46185da2b55.png">

**Reply by Hiro-Tokyo** [reply]
Hi Josh, when I see distance on phone, font sizes a little bit big. I can put CSS to change font size but if you have any better way, please let me know.
<img width="318" alt="Screenshot 2023-03-09 at 10 36 46" src="https://user-images.githubusercontent.com/100449629/223893252-6d8e3954-ef00-4bbf-8f35-8e2a6c42b97f.png">

**Reply by joshreisner** [reply]
hi hiro, yes there is a problem here. i'll fix it soon

**Reply by joshreisner** [reply]
hi hiro, let me know if this looks correct: https://deploy-preview-300--tsml-ui.netlify.app/tests/japan.html?mode=me

**Reply by Hiro-Tokyo** [reply]
Thanks Josh! It's beautiful!![image](https://user-images.githubusercontent.com/100449629/223933622-9e3871a0-b4f6-4ba4-bdec-d64aa332efd3.jpeg)

**Reply by Hiro-Tokyo** [reply]
Hi Josh, Japan Online Intergroup is using CSV importing with the plugin. If we would like to connect Google Sheet directly with the plugin, can we use this importer? I think I'm not understanding this setting.

https://sheets.code4recovery.org

<img width="1294" alt="Screenshot 2023-03-16 at 8 56 32" src="https://user-images.githubusercontent.com/100449629/225470710-a81f50c1-3e4a-4664-acbe-f1e6efacb6a9.png">

**Reply by joshreisner** [reply]
i'm sorry, i don't understand the question, could you rephrase? why do they prefer a CSV and TSML over a Google Sheet and TSML UI?

**Reply by Hiro-Tokyo** [reply]
I mean it's just my curious. Is there way to connect Google Sheet directly with 12 step meeting list plugin? They are using  12 step meeting list plugin with Wordpress. I mean how we can use Import Data Source setting in the plugin.

Kanasai area is using TSML UI with the code you gave me.

**Reply by Hiro-Tokyo** [reply]
Probably I'm not understanding how I can use Import Data Source setting.

**Reply by joshreisner** [reply]
ok thanks. yes technically it is possible to import a google sheet into TSML, but it is a little tricky to get it set up. and it doesnt sound like they need to? if they are using a Google Sheet and TSML UI and they are fine then they could uninstall the plugin and just use it that way.

if they want to combine the meetings from two different sources (for example two Sheets, or one Sheet and WordPress meetings) then it would make sense.

also if their site is getting such high traffic that it is running into errors with their API key, then they might also consider importing, or using the sheets.code4recovery.org project. in either scenario the benefit would be that they could serve a more users because the data would be imported and saved as a JSON file which doesn't require an API key. the tradeoff would be that after making changes in the Sheet they would need to reimport / rebuild the JSON file for users to see changes

**Reply by Hiro-Tokyo** [reply]
Thanks Josh! A Google Sheet and TSML UI is simple way! I have one more question. You gave me the code this below. Is there any way for TSML UI? A reason why I'm asking is that Online Intergroup is having same issue continuously. Some meetings do not want to post online meeting URLs on website.


if (!empty($tsml_conference_providers)) {
	$tsml_conference_providers += [
		'meet.jit.si' => 'Jitsi',
		'myrandom.website.jp' => 'One Day at a Time Group,
	];
}

**Reply by joshreisner** [reply]
hi hiro, we typically discourage this (to prioritize the experience of the user / member looking for a meeting), but yes you can do it by extending your configuration object: 

```html
<script>
  var tsml_react_config = {
    distance_unit: 'km',
    conference_providers: {
      'sites.google.com': 'Custom',
      'tinyurl.com': 'Custom',
    },
  };
</script>
```

**Reply by joshreisner** [reply]
you can change the domain to be whatever they need to use, and "Custom" to be whatever you want the button to say

**Reply by Hiro-Tokyo** [reply]
Thank you so much! I understand what you say but the intergroup is talking with some meetings so it would be helpful!

**Reply by Hiro-Tokyo** [reply]
Hi Josh, I have a question. I don't see 'Group Phone' column on TSML UI Google Sheet. It is still available?

**Reply by Hiro-Tokyo** [reply]
I tried make a column then put tel: command. It works but if any other way, please let me know.

**Reply by joshreisner** [reply]
You can add a `Phone` column - that will show up as a tel link automatically. In the field, just use numbers, and + and ,. You can use a `'` symbol in front of the number to treat it as text.

Example: https://tsml-ui.code4recovery.org/tests/japan.html?meeting=shonan-friday

I added the column as an example to this sheet: https://docs.google.com/spreadsheets/d/1S18u8LyDXJy-nkDbAi321imElV8JNXR3R8ho0P0srHE/edit#gid=0

**Reply by Hiro-Tokyo** [reply]
Thanks Josh! I did that way.  Phone symbol shows up on website page but does not connect to call. Makes this link, https://aajapan-online-intergroup.org/index.php/meeting-list-new/080-4112-5325

**Reply by Hiro-Tokyo** [reply]
<img width="414" alt="Screenshot 2023-03-30 at 14 51 53" src="https://user-images.githubusercontent.com/100449629/228742084-3f9eb530-b3d8-4760-b8f5-c4c70a5a8fb8.png">
<img width="415" alt="Screenshot 2023-03-30 at 14 52 04" src="https://user-images.githubusercontent.com/100449629/228742087-0c810b38-bc23-498a-8e32-5cd5fd29d850.png">

**Reply by Hiro-Tokyo** [reply]
<img width="528" alt="Screenshot 2023-03-30 at 14 58 49" src="https://user-images.githubusercontent.com/100449629/228743129-e85976ce-5652-4056-8c8e-23d6534ae8e8.png">

**Reply by joshreisner** [reply]
oops i think it is a bug - just pushed a fix 🛠️

**Reply by Hiro-Tokyo** [reply]
Thanks!!!

**Reply by Hiro-Tokyo** [reply]
Hi Josh, I would like to just make sure that Japan Online Intergroup could make meeting information column wider on PC view as the intergroup does not use map. If there is any issue, please let me know. I used CSS customize.
<img width="722" alt="Screenshot 2023-03-31 at 12 55 15" src="https://user-images.githubusercontent.com/100449629/229019164-258e0fc5-eab2-442f-ac7b-fa41ba45c545.png">

**Reply by joshreisner** [reply]
hi hiro, sounds good!

**Reply by Hiro-Tokyo** [reply]
Thanks! and seems Tel Link issue has improved!

**Reply by Hiro-Tokyo** [reply]
Hi Josh! When we would like to use Google Map instead of Map Box, how we have to do?

<div
  id="tsml-ui"
  data-src="https://docs.google.com/spreadsheets/d/your.google.sheet.id/edit#gid=more.stuff.here"
  data-google="your.google.api.key"
  data-mapbox="pk.your.mapbox.access.token.goes.here"
  data-timezone="Asia/Tokyo"
></div>
<script src="https://tsml-ui.code4recovery.org/app.js" async></script>

**Reply by joshreisner** [reply]
hi Hiro, TSML UI only supports mabox, sorry 😞

**Reply by Hiro-Tokyo** [reply]
Okay I understood! Thanks Josh!

**Reply by Hiro-Tokyo** [reply]
I have one more question. Is there any way group name from the group column on google sheet is listed on meeting info? I mean column 'E'.

![image](https://github.com/code4recovery/12-step-meeting-list/assets/100449629/0af74a8a-8249-45c0-8637-25718a11c46d)

**Reply by Hiro-Tokyo** [reply]
I saw if group has multiple meetings, it show up but if group has only one meeting, not show up so I'm asking.

**Reply by joshreisner** [reply]
yes, if it's just a group name it looks funny all by itself. there are a number of columns that, if filled, should make the group section show up:

* Group Notes
* District
* Email
* Website
* Phone
* Venmo
* Square
* PayPal

**Reply by Hiro-Tokyo** [reply]
Thanks! So there is 'Group Notes' column aside from 'Notes' column?

**Reply by joshreisner** [reply]
Yes, there can be if you want to add it. Notes is for things about the meeting, Group Notes can be used to add some descriptive text below the group name, and can be for things about the group.

Column order does not matter, so you can insert the column anywhere in the sheet

**Reply by Hiro-Tokyo** [reply]
Thanks! It's working well! We tried group name + district!

**Reply by Hiro-Tokyo** [reply]
![image](https://github.com/code4recovery/12-step-meeting-list/assets/100449629/9be5cedd-d3cc-49bb-94f9-54d911c4ed77)

**Reply by Hiro-Tokyo** [reply]
I sent pull request as I made some Japanese translation changes. Please make sure that.

And if you can, could you explain how to compile after merge?

https://github.com/code4recovery/tsml-ui#contributing

**Reply by joshreisner** [reply]
thanks hiro! it would be good to build assets as part of this PR. please

1. install NPM if it is not installled
2. run `npm i` in the project folder
3. run `npx mix --production` in the project folder

i have expanded the section of the readme about developing if you would like to preview your changes locally: https://github.com/code4recovery/tsml-ui#contributing
